encoding/descaling.md

```diff
@@ -1,477 +1,332 @@
+by zooming in 4x with Nearest Neighbor
+and switching between the frames
+with the left and right arrow buttons.
+You can enable the zoom here:
 
-This is a nice frame to use as reference.
-The background is a bit blurry,
-but it isn't full of effects
-and is fairly bright.
-The lineart is very clear.
+![](images/descale/descale_zoom_a.png)
 
-We will now make use of the getnative.py script
-to figure out what resolution
-this anime was produced at.
-Run the following in your terminal:
+Clip "No zoom" and select "Fixed ratio".
+You can set the zoom level next to it,
+and the scaler used next to that.
+Make sure to set it to Nearest!
 
-```bash
-$ python getnative.py "descale_manaria04.png"
-```
+Figuring out what to look out for
+is a case of trial and error.
+The best thing to keep an eye on
+is lineart as well as small detail like stars.
+It isn't uncommon for noise to become weaker after a rescale
+(since that's usually added after upscaling, too),
+so try not to be too distracted by that.
 
-It should show the following:
+[descale]: https://github.com/Irrational-Encoding-Wizardry/vapoursynth-descale
+[fmtc]: https://forum.doom9.org/showthread.php?t=166504
 
-```
-Using imwri as source filter
-501/501
-Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33
-Native resolution(s) (best guess): 878p
-done in 18.39s
-```
 
-If you check the directory
-where you executed the script,
-you will find a new folder
-called "getnative".
-You can find the following graph
-in there as well:
-
-![Manaria Friends — 01 (frame 4 getnative graph)](images/descale_graph.png)
-
-
-The X-axis shows the resolutions that were checked,
-and the Y-axis shows the relative error.
-The relative error refers to
-the difference between the original frame
-and the rescaled frame.
-What you're looking for are
-the spikes that show a low relative error.
-In this case it very clearly points to 878p.
-
-As a sidenote,
-it's important to keep in mind that this script
-can't find native 1080p elements.
-This is because it descales the frame
-and re-upscales it afterwards
-to determine the relative error.
-You can't descale to 1080p
-if the frame is already 1080p.
-If you have reason to believe
-that your show might be native 1080p,
-you've got to go with your gut.
-
-![Date A Live III — 01 (getnative graph)](images/descale_native1080_graph.png)
-
-An example of a graph for a native 1080p show.
-
-You may notice that the line swerves a bit
-in the first graph.
-There are going to be cases where
-you will get odd graphs like these,
-so it's important to know
-when the results are safe enough to descale
-or when they're too risky.
-Here is an example of
-a "bad" graph:
-
-![Miru Tights — 02 (getnative graph)](images/descale_bad_graph1.png)
+# Re-scaling
 
-```
-Output:
-Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33
-Native resolution(s) (best guess): 869p, 848p
-```
+A fairly common practice is
+to upscale the descaled clip to a standard resolution whenever possible
```

Comment:
"whenever possible" is ambiguous. When is it possible? What could prevent one from resizing to a standard resolution?